src/sentry/api/endpoints/project_user_reports.py

```diff
@@ -50,9 +50,18 @@ def get(self, request, project):
         :pparam string project_slug: the slug of the project.
         :auth: required
         """
+        try:
+            environment = self._get_environment_from_request(
+                request,
+                project.organization_id,
+            )
+        except Environment.DoesNotExist:
+            return Response({'environment': 'Invalid environment'}, status=400)
+
         queryset = UserReport.objects.filter(
             project=project,
             group__isnull=False,
+            environment=environment,
```

Comment:
I _think_ we want to not do this if `environment is None`.

If `environment` is None from above, you're going to be filtering and finding only the ones that _don't_ have an environment set. Whereas, I'm pretty sure our definition of "no environment" means, "return all environments". Not "untagged". If that makes sense.

cc @tkaemming

Reply:
Yes, this is correct. The filter should only be applied if `environment is not None` (since `None` signifies "no filter value" in this context.)